common/lib/xmodule/xmodule/js/src/video/06_video_progress_slider.js

```diff
@@ -235,8 +222,15 @@ function () {
     }
 
     function updatePlayTime(params) {
-        var time = Math.floor(params.time),
-            duration = Math.floor(params.duration);
+        var time = Math.floor(params.time);
+        // params.duration could accidentally be construed as a floating
+        // point double. Since we're displaying this number, round down
+        // to nearest second
+        var duration = Math.floor(params.duration);
+
+        if (this.config.endTime !== null) {
+            duration = Math.min(this.config.endTime, duration);
+        }
```

Comment:
you're right, this is legacy naming and something that needs to be fixed. Will submit an update to this